packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,6 +336,53 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
 
     private createTiptapEditor(): Editor {
         const customLink = this.getCustomLinkExtension();
+        const validAbsoluteLinkRegex = /^https?:\/\//i;
+
+        /**
+         * @param htmlString contains the html string of the copied content. If the content is a link, the `htmlString` contains anchor tag and a href value.
+         * ProseMirror reference for `transformPastedHTML`: https://prosemirror.net/docs/ref/#view.EditorProps.transformPastedHTML
+         */
+        const transformPastedHTML = (htmlString: string): string => {
+            const templateElement = document.createElement('template');
+            templateElement.innerHTML = htmlString;
+            const templateDocument = templateElement.content.ownerDocument;
+
+            templateElement.content
+                .querySelectorAll('a')
+                .forEach(anchorElement => {
+                    let tempAnchorElement: HTMLAnchorElement | Anchor = anchorElement;
+                    if (
+                        anchorElement.parentElement
+                        && anchorElement.parentElement.tagName === 'NIMBLE-ANCHOR'
+                    ) {
+                        tempAnchorElement = anchorElement.parentElement as Anchor;
+                    }
+                    const href = anchorElement.getAttribute('href');
+                    // When pasting a link, the `href` attribute of the anchor element should be a valid HTTPS/HTTP link;
+                    // else, simply rendered as a plain text in the same node by creating a span element and replace it with anchor element.
+                    if (href && validAbsoluteLinkRegex.test(href)) {
+                        tempAnchorElement.textContent = href; // Modifying the anchor element text content with its href
+                    } else {
+                        const spanElement = templateDocument.createElement('span');
+                        spanElement.textContent = tempAnchorElement.textContent;
+                        tempAnchorElement.replaceWith(spanElement);
+                    }
+                });
+
+            return templateElement.innerHTML;
+        };
+
+        // To prevent the pasting of files/images into the editor and avoid making any network requests.
+        // ProseMirror handlePaste: https://prosemirror.net/docs/ref/#view.EditorProps.handlePaste
+        // The below function needs an update when image support is added
+        const handlePaste = (
+            _view: EditorView,
+            event: ClipboardEvent
+        ): void => {
+            if (event.clipboardData && event.clipboardData.files.length >= 1) {
```

Comment:
Are you confident that everything that could result in a side effect (like making network requests) will be represented by a file in the `clipboardData`?

Reply:
We have completely removed this method as getting data:url is not an issue as we discussed over a call yesterday.